faker/providers/internet/__init__.py

```diff
@@ -99,15 +101,23 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self):
-        """
-        Convert 32-bit integer to dotted IPv4 address.
-        """
-        return ".".join(map(lambda n: str(random.randint(-2147483648, 2147483647) >> n & 0xFF), [24, 16, 8, 0]))
-
-    def ipv6(self):
-        res = [hex(random.randint(0, 65535))[2:].zfill(4) for i in range(0, 8)]
-        return ":".join(res)
+    def ipv4(self, network=False):
+        """ Produce a random IPv4 address or network with a valid CIDR. """
+        address = str(ip_address(random.randint(
+            0, (2 ** IPV4LENGTH) - 1)))
+        if network:
+            address += '/' + str(random.randint(0, IPV4LENGTH))
+            address = str(ip_network(unicode(address), strict=False))
```

Comment:
because of the concatenation on the line above, I would expect `address` to already be coerced to a string at this point, making `unicode()` unnecessary